src/MutableIterator.php

```diff
@@ -0,0 +1,114 @@
+<?php
+
+/**
+ *
+ * @category   Humbug
+ * @package    Humbug
+ * @copyright  Copyright (c) 2015 Pádraic Brady (http://blog.astrumfutura.com)
+ * @license    https://github.com/padraic/humbug/blob/master/LICENSE New BSD License
+ * @author     Thibaud Fabre
+ */
+namespace Humbug;
+
+use Symfony\Component\Finder\Finder;
+use Traversable;
+
+class MutableIterator implements \IteratorAggregate, \Countable
+{
+
+    /**
+     * @var Container
+     */
+    private $container;
+
+    /**
+     * @var Finder
+     */
+    private $finder;
+
+    /**
+     * @var array
+     */
+    private $mutables = null;
+
+    /**
+     * @param Container $container
+     * @param string[] $directories
+     * @param string[] $excludes
+     * @param string[] $names
+     */
+    public function __construct(Container $container, $directories, $excludes, $names)
+    {
+        $this->container = $container;
+        $this->finder = $this->prepareFinder($directories, $excludes, $names);
```

Comment:
Agreed with both of you in the end. MutableFinder is probably overkill considering it wont add any value to Finder, and it makes sense to pass in the prepared finder to the MutableIterator class.